lms/static/js/spec/student_account/account_settings_fields_spec.js

```diff
@@ -0,0 +1,325 @@
+                    // But if we change the message, it should not get reset.
+                    view.message("Do not reset this!");
+                    jasmine.Clock.tick(5000);
+                    expectMessageContains(view, "Do not reset this!");
+                }
+            });
+
+            it("sends a PATCH request when saveAttributes is called", function() {
+
+                requests = AjaxHelpers.requests(this);
+
+                var fieldViewClass = FieldViews.FieldView;
+                var fieldData = createFieldData(fieldViewClass, {
+                    title: 'Preferred Language',
+                    valueAttribute: 'language',
+                    helpMessage: 'Your preferred language.'
+                })
+
+                var view = new fieldViewClass(fieldData);
+                view.saveAttributes(
+                    {'language': 'ur'},
+                    {'headers': {'Priority': 'Urgent'}}
+                );
+
+                var request = requests[0];
+                expect(request.method).toBe('PATCH');
+                expect(request.requestHeaders['Content-Type']).toBe('application/merge-patch+json;charset=utf-8');
+                expect(request.requestHeaders['Priority']).toBe('Urgent');
+                expect(request.requestBody).toBe('{"language":"ur"}');
+            });
+
+            it("correctly renders ReadonlyFieldView", function() {
+                var fieldData = createFieldData(FieldViews.ReadonlyFieldView, {
+                    title: 'Username',
+                    valueAttribute: 'username',
+                    helpMessage: 'The username that you use to sign in to edX.'
+                });
+                var view = new FieldViews.ReadonlyFieldView(fieldData).render();
+
+                expectTitleAndMessageToBe(view, fieldData.title, fieldData.helpMessage);
+                expect(view.$('.u-field-value input').val().trim()).toBe(USERNAME);
+            });
+
+            it("correctly updates ReadonlyFieldView on model update", function() {
+                var fieldData = createFieldData(FieldViews.ReadonlyFieldView, {
+                    title: 'Username',
+                    valueAttribute: 'username',
+                    helpMessage: 'The username that you use to sign in to edX.'
+                });
+                var view = new FieldViews.ReadonlyFieldView(fieldData).render();
+
+                expect(view.$('.u-field-value input').val().trim()).toBe(USERNAME);
+                view.model.set({'username': 'bookworm'});
+                expect(view.$('.u-field-value input').val().trim()).toBe('bookworm');
+            });
+
+            it("correctly renders TextFieldView", function() {
+                var fieldData = createFieldData(FieldViews.TextFieldView, {
+                    title: 'Full Name',
+                    valueAttribute: 'name',
+                    helpMessage: 'This is the name used on your edX certificates. Changes to this field are reviewed.'
```

Comment:
Question left over from previous review: Who's reviewing the changes to this field, and why? I'm sure end users will wonder about this too.